hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/offload/MapStoreOffloadingBouncingNodeTest.java

```diff
@@ -0,0 +1,321 @@
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hazelcast.map.IMapAccessors.getPendingOffloadedOpCount;
+import static com.hazelcast.spi.impl.operationservice.impl.OperationServiceAccessor.getAsyncOperationsCount;
+import static java.util.Arrays.asList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParametrizedRunner.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParallelParametersRunnerFactory.class)
+@Category({ParallelJVMTest.class, QuickTest.class})
+public class MapStoreOffloadingBouncingNodeTest extends HazelcastTestSupport {
+
+    @Parameterized.Parameters(name = "writeBehindEnabled: {0}")
+    public static Collection<Object[]> parameters() {
+        return asList(new Object[][]{
+                {true},
+                {false}
+        });
+    }
+
+    @Parameterized.Parameter
+    public boolean writeBehindEnabled;
+
+    private static final int TEST_RUN_SECONDS = 20;
+
+    protected Config getConfig(String mapName) {
+        Config config = smallInstanceConfigWithoutJetAndMetrics();
+        config.getMapConfig(mapName)
+                .setBackupCount(1)
+                .setAsyncBackupCount(0)
+                .setInMemoryFormat(getInMemoryFormat())
+                .getMapStoreConfig()
+                .setEnabled(true)
+                .setOffload(true)
+                .setWriteDelaySeconds(writeBehindEnabled ? 3 : 0)
+                .setImplementation(new MapStoreAdapter() {
+
+                    @Override
+                    public void store(Object key, Object value) {
+                        sleepRandomMillis();
+                        super.store(key, value);
+                    }
+
+                    @Override
+                    public void delete(Object key) {
+                        sleepRandomMillis();
+                        super.delete(key);
+                    }
+
+                    @Override
+                    public Object load(Object key) {
+                        sleepRandomMillis();
+                        return randomStringOrNull();
+                    }
+                });
+        return config;
+    }
+
+    protected InMemoryFormat getInMemoryFormat() {
+        return InMemoryFormat.BINARY;
+    }
+
+    private static Object randomStringOrNull() {
+        return RandomPicker.getInt(0, 10) < 2
+                ? null : randomString();
+    }
+
+    private static void sleepRandomMillis() {
+        sleepMillis(RandomPicker.getInt(0, 3));
+    }
+
+    @Test(timeout = 5 * 60 * 1000)
+    public void stress() throws InterruptedException {
```

Comment:
Do we need `TransactionsWithWriteBehind_whenNoCoalescingQueueIsFullTest.stress()` to exist with this new test?

Reply:
Transactional api is only tested in that test. Other one is for non-txn api. Let's keep both.